backend/app/Http/Controllers/ProjectController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Project;
+use Illuminate\Http\Request;
+
+class ProjectController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        $projects = Project::with('user')->paginate(10);
+
+        return response()->json([
+            'success' => true,
+            'data' => $projects,
+            'message' => 'Projects retrieved successfully.'
+        ], 200);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        $validatedData = $request->validate([
+            'title' => 'required|string|max:255',
+            'description' => 'required|string',
+            'stack' => 'required|string'
+        ]);
+
+        $project = auth()->user()->projects()->create($validatedData);
+
+        return response()->json([
+            'success' => true,
+            'data' => $project,
+            'message' => 'Project created successfully.'
+        ], 201);
+    }
+
+    /**
+     * Display the specified resource.
+     */
+    public function show(Project $project)
+    {
+        $project->load('user');
+
+        return response()->json([
+            'success' => true,
+            'data' => $project,
+            'message' => 'Project retrieved successfully.'
+        ], 200);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request, Project $project)
+    {
+        // 403 Forbidden. The user does not have permissions to edit this project.
+        if (auth()->id() !== $project->user_id) {
```

Comment:
**nitpick:** could we also add a validation check for when `auth()->id()` might be null? Except if we do not envisage a scenario where an unauthenticated user would have access to the update project call then we are good.

Reply:
I'd say that case scenario is out of the scope of this PR. If we want to allow anonymous projects, we would need to decide who and how they can update them. I'd say we allow only authenticated users for now, unless there's a clear path on how to do the unauthenticated thing.